query/transformation.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package query contains all the logic to parse
+// and execute queries against the underlying metric system.
+package query
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/square/metrics/api"
+)
+
+// A transformation takes the list of values, and a "scale" factor:
+// It is the unitless quotient (sample duration / 1s)
+// So if the resolution is "once every 5 minutes" them we get a scale factor of 5*60 = 300
+type transformation func([]float64, transformationParameter) ([]float64, error)
+
+type transformationParameter struct {
```

Comment:
My concern was that other parameters might be implemented in the future (like providing the `Timerange` explicitly) and if they were given individually then every already-defined transformation which didn't use them would have to change.

If we can set the parameters in stone then it would be a better idea to pass them individually.